exporter/prometheusexporter/README.md

```diff
@@ -34,4 +39,46 @@ exporters:
     metric_expiration: 180m
     resource_to_telemetry_conversion:
       enabled: true
+    prometheus_convention: true
+    rename:
+      "hw.temperature.temperature_celsius": "hw_temperature_celsius"
+      "system.io": "system_io_bytes_total"
 ```
+
+## Metrics Renaming
+
+[OpenTelemetry's metric semantic convention](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/README.md) is not compatible with [Prometheus' own metrics naming convention](https://prometheus.io/docs/practices/naming/). It is therefore recommended to rename OpenTelemetry metrics before exporting them to Prometheus.
+
+This Prometheus Exporter ensures the exported metrics follow Prometheus' naming convention when the `prometheus_convention` configuration option is set to `true`.
+
+> Notes:
+>
+> - Even if `prometheus_convention` is not set to `true`, the exporter will make sure the exported metrics contain only valid characters and will notably replace non-alphanumeric characters with underscores.
+> - Any metrics renaming specified under the `rename` configuration option takes precedence over the automatic renaming described below. This allows the user to cover some edge cases "manually".
+> - Namespace prepending specified with the `namespace` option is performed **after** all renaming operations
+
+List of transformations performed on OpenTelemetry metrics:
+
+| Case                                             | Transformation                                                                                                                   | Example                                                                             |
+| ------------------------------------------------ | -------------------------------------------------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------------- |
+| Standard unit                                    | Convert the unit from [Unified Code for Units of Measure](http://unitsofmeasure.org/ucum.html) to Prometheus standard and append | `system.filesystem.usage` with unit `By` ==> `system_filesystem_usage_bytes`        |
+| Non-standard unit (unit is surrounded with `{}`) | Remove brackets and append, if not already mentioned in the metric name                                                          | `system.network.dropped` with unit `{packets}` ==> `system_netowrk_dropped_packets` |
```

Comment:
That's not what I understand from OpenMetrics:

> If non-base units can not be avoided and conversion is not feasible, the actual unit should still be included in the metric name for clarity.

So non-standard unit should be appended to the metric name (if not already present in the metric name).

Reply:
Okay, I think I just understood: for non-standard units, the unit must be appended to the metric name (if not already present) and the `Unit` field emptied so that it doesn't end up in `# UNIT ...` when exported. Am I correct?

Reply:
Okay, to summarize:

| Otel Metric Name | Unit | Prometheus Metric |
|---|---|---|
| `astro.galaxy.distance` | `m` | `astro_galaxy_distance_meter` |
| `astro.galaxy.distance` | `ly` ([c/s notation for light-years](https://ucum.org/ucum.html#const)) | `astro_galaxy_distance_ly` |
| `router.network.udp_traffic` | `{packets}` | `router_network_udp_traffic` |

What about broken cases (receivers are not all equal, and some metrics may not follow Otel's semantic convention?

| Otel Metric Name | Unit | Prometheus Metric |
|---|---|---|
| `router.network.udp_traffic` | `packets` (no brackets) | `router_network_udp_traffic_packets` or `router_network_udp_traffic`? |
| `router.network.bandwidth_utilization` | `percent` | `router_network_bandwidth_utilization_percent`? |
|`router.network.bandwidth_utilization` | `%` | `router_network_bandwidth_utilization`? |

Reply:
For broken cases, we should probably assume it is a nonstandard unit we haven't heard of, and treat it the same as you do `ly` above.  For units that include characters that aren't allowed in metric names, we can either drop them entirely, or just drop the invalid characters.  I'm inclined to start by dropping the unit entirely in that case.